csharp/src/Microsoft.ML.OnnxRuntime/ManagedProjections.shared.cs

```diff
@@ -19,48 +19,16 @@ namespace Microsoft.ML.OnnxRuntime
     /// For example, a sequence instance would contain a list of NamedOnnxValue instances
```

Comment:
Ugh. That's just misleading as there's no concept of a name being attached to the individual elements in the sequence. Seems like we could have put anything in the collection given NamedOnnxValue::_value is an `Object` and the choice to use `NamedOnnxValue` for sequence elements in `DisposableNamedOnnxValue::FromNativeSequence` was incorrect.

Reply:
Okay, then we need another structure for nested elements to hold the type. What you see is already a part of the previous release.  Because `_value` is an object, we need to some known type to cast to and read the `OnnxValueType` of the contained object, and that is the function of the nested `NamedOnnxValue` in this case.

It is not a problem with `OrtValue`, since we know that OrtValues  can only contain other `OrtVlaues` and we can read the type of it.

What would be your suggestion?